packages/live-share/src/LivePresence.ts

```diff
@@ -399,68 +426,56 @@ export class LivePresence<TData extends object = object> extends DataObject<{
     }
 
     private updateMembersListWithInfo(
-        evt: ILivePresenceEvent<TData>,
+        evt: LivePresenceReceivedEventData<TData>,
         local: boolean,
         info: IClientInfo
     ): void {
         const emitEvent = (user: LivePresenceUser<TData>) => {
+            this._lastEmitPresenceStateMap.set(user.userId, user.state);
             this.emit(LivePresenceEvents.presenceChanged, user, local);
             if (local) {
-                this._logger.sendTelemetryEvent(
+                this._logger?.sendTelemetryEvent(
                     TelemetryEvents.LivePresence.LocalPresenceChanged,
                     { user: evt }
                 );
             } else {
-                this._logger.sendTelemetryEvent(
+                this._logger?.sendTelemetryEvent(
                     TelemetryEvents.LivePresence.RemotePresenceChanged,
                     { user: evt }
                 );
             }
         };
 
-        let pos = 0;
-        const userId = info.userId;
-        for (; pos < this._users.length; pos++) {
-            const current = this._users[pos];
-            const cmp = userId.localeCompare(current.userId);
-            if (cmp == 0) {
+        let isNewUser = true;
+        for (let pos = 0; pos < this._users.length; pos++) {
+            const checkUser = this._users[pos];
+            if (info.userId === checkUser.userId) {
                 // User found. Apply update and check for changes
-                if (current.updateReceived(evt, info)) {
-                    emitEvent(current);
+                if (checkUser.updateReceived(evt, info)) {
+                    emitEvent(checkUser);
                 }
-
-                return;
-            } else if (cmp > 0) {
-                // New user that should be inserted before this user
-                break;
+                isNewUser = false;
+            } else if (
```

Comment:
Asking for understanding purposes. Instead of looping only until the user is found, we loop through all of them to make sure to emit events when any user goes offline?

Reply:
Yeah exactly. We can probably remove this once we integrate `IAzureAudience`, but without this the developer has to set up some background loop and poll the list to see who went offline, or do something like `presence.getUsers()` and loop through on every presence change, which only helps if the checked user here actually changed anyways.